trino/sqlalchemy/dialect.py

```diff
@@ -299,18 +325,13 @@ def get_indexes(self, connection: Connection, table_name: str, schema: str = Non
 
         partitioned_columns = None
         try:
-            partitioned_columns = self._get_columns(connection, f"{table_name}$partitions", schema, **kw)
+            partitioned_columns = self._get_partitions(connection, f"{table_name}", schema)
         except Exception as e:
             # e.g. it's not a Hive table or an unpartitioned Hive table
             logger.debug("Couldn't fetch partition columns. schema: %s, table: %s, error: %s", schema, table_name, e)
         if not partitioned_columns:
             return []
-        partition_index = dict(
-            name="partition",
-            column_names=[col["name"] for col in partitioned_columns],
-            unique=False
-        )
-        return [partition_index]
+        return partitioned_columns
```

Comment:
please return `ReflectedIndex`. Superset (for which we added this `get_indexes` method) relies on the output being SQLAlchemy format.